### PR TITLE
Feature/indicator variations

### PR DESCRIPTION
**Pull request by agnessa**
Puts in place schema changes related to indicators & time series values and adapts the time series upload process accordingly.

New columns in the indicators table:
- `unit_of_entry` and `conversion_factor` to allow managing unit conversions on indicator level
- `subcategory` and `stackable_subcategory` instead of the earlier `stack_family`
- `parent_id` to link with master indicator

New columns in the time_series_values table:
- `unit_of_entry` and `conversion_factor` to allow specifying unit conversions on time series values level

The time series upload process has been adapted to match the current template. Additionally, unit conversions are no longer performed before storing the data, instead the value and unit of entry is stored as provided as long as the unit of entry matches either the standard unit or unit of entry defined in the indicator - otherwise an error is returned.

Model permissions are now checked in the upload process.

The uploaded file handling logic has been simplified, which hopefully will fix this feature on staging.

It is now possible to overwrite old data by reuploading.

### CODE REVIEW

**Comment by simaob** [comment]
While running the specs I saw some deprecation warnings regarding the use of `each_with_index`

> DEPRECATION WARNING: Method each_with_index is deprecated and will be removed in Rails 5.1, as `ActionController::Parameters` no longer inherits from hash. Using this deprecated behavior exposes potential security problems. If you continue to use this method you may be creating a security vulnerability in your app that can be exploited. Instead, consider using one of these documented methods which are not deprecated: http://api.rubyonrails.org/v5.0.2/classes/ActionController/Parameters.html (called from fetch_all at /Users/Simao/dev/emissions-scenario-portal/app/models/team.rb:24)

In case you have your warnings turned off :D

File: app/views/indicators/index.html.erb [hunk 1]
```diff
@@ -99,14 +99,14 @@
...
-            <div class="c-table-list-item__text f-ff1-m"><%= indicator.stack_family %></div>
+            <div class="c-table-list-item__text f-ff1-m"><%= indicator.subcategory %></div>
           </div>
           <div class="small-1 columns">
             <div class="c-table-list-item__text f-ff1-m"><%= indicator.unit %></div>
           </div>
           <div class="small-4 columns">
             <div class="c-table-list-item__text f-ff1-m">
-              <%= indicator.definition.length > 120 ? truncate(indicator.definition, length: 120) : indicator.definition %>
+              <%= indicator.definition && indicator.definition.length > 120 ? truncate(indicator.definition, length: 120) : indicator.definition %>
```
Comment (simaob) [inline app/views/indicators/index.html.erb]: I'm wondering if the length check is necessary here. I believe that the `truncate` method will return the untouched string if its length is <= than the passed length parameter.

File: app/views/scenarios/show.html.erb [hunk 2]
```diff
@@ -103,7 +103,7 @@
           </div>
           <div class="small-5 columns">
             <div class="c-table-list-item__text f-ff1-m">
-              <%= indicator.definition.length > 150 ? truncate(indicator.definition, length: 150) : indicator.definition %>
+              <%= indicator.definition && indicator.definition.length > 150 ? truncate(indicator.definition, length: 150) : indicator.definition %>
```
Comment (simaob) [inline app/views/scenarios/show.html.erb]: Same as above comment about the `truncate` method usage.

File: app/views/indicators/index.html.erb (hunk 1 above)
Comment (agnessa) [inline app/views/indicators/index.html.erb]: here goes a badge, thank you: 🏆

**Comment by agnessa** [comment]
select2 got fixed after rebasing with develop

**Comment by agnessa** [comment]
About the file, I tested with the latest template (Time_series_GCAM.csv). Amended the sample 'Please follow this structure' file.

**Pull request merged**